modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridNioServer.java

```diff
@@ -2860,14 +2895,17 @@ final void reset0() {
      * Gets outbound messages queue size.
      *
      * @return Write queue size.
+     * @deprecated Will be removed in the next major release and replaced with new metrics API.
      */
+    @Deprecated
     public int outboundMessagesQueueSize() {
-        int res = 0;
-
-        for (GridSelectorNioSessionImpl ses : sessions)
-            res += ses.writeQueueSize();
+        if (mreg == null)
+            return -1;
 
-        return res;
+        return (int) mreg.longAdderMetric(
```

Comment:
We should store `LongMetric` inside object instance.

Reply:
It's about unnecessary `Map#get` operation.
Every 'mreg.longAdderMetric` translates to `Map#get`.
Why do we need it? We can store the variable, isn't it?